src/types/alerts.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  DeadmanCheck,
+  ThresholdCheck as ThresholdCheckGen,
+  GreaterThreshold as GreaterThresholdGen,
+  LesserThreshold as LesserThresholdGen,
+  RangeThreshold as RangeThresholdGen,
+  SlackNotificationRule,
+  PagerDutyNotificationRule,
+  SMTPNotificationRule,
+} from '../api'
+
+export interface ThresholdCheck extends ThresholdCheckGen {
+  thresholds: Array<
+    GreaterThresholdGen | LesserThresholdGen | RangeThresholdGen
+  >
+}
+
+export type Check = DeadmanCheck | ThresholdCheck
```

Comment:
You're right, even though our generator can not make use of the discriminator, it would be better practice to include it. I'm going to try changing the swagger and see whether that breaks anything else.

Reply:
also added more explicit types